syz-manager/manager.go

```diff
@@ -1066,7 +1064,170 @@ func (mgr *Manager) fuzzerConnect() ([]rpctype.RPCInput, BugFrames, bool) {
 	for frame := range mgr.dataRaceFrames {
 		dataRaceFrames = append(dataRaceFrames, frame)
 	}
-	return corpus, BugFrames{memoryLeaks: memoryLeakFrames, dataRaces: dataRaceFrames}, mgr.coverFilterFilename != ""
+	var err error
+	once.Do(func() {
+		err = mgr.updateKernelModules(a)
+		if err != nil {
+			return
+		}
+		mgr.ready = true
+		coverFilterBitmap, coverFilter, err = mgr.updateCoverFilter()
```

Comment:
Don't use global variables. These should be fields of Manager type.
Also sync.Once makes sense for potential concurrent calls, here we synchronize everything with mgr.mu, so a plain bool modulesInitialized will do much better (or maybe just use some manager field that we initialize in updateKernelModules).

Reply:
not sure if I understand correctly, please check again in new patches